hw2/03-count.js

```diff
@@ -1,3 +1,35 @@
 // Add your code here
+const input = document.querySelector("input");
+let searchArray = [];
+//const mainTextArray = cleanText();
+let paragraph = document.querySelector("#paragraph");
 
-input.addEventListener('keydown', handleKeyDown);
+let text = document.getElementById("paragraph").innerHTML.toLowerCase();
+input.addEventListener("keydown", handleKeyDown);
+
+function handleKeyDown(e) {
+  if (e.key === "Backspace") {
+    e.target.value = "";
+    searchArray = [];
+  } else if (e.key === "Shift") {
+    let searchStr = searchArray.toString();
+    searchStr = searchStr.replaceAll(",", "");
+    findMatch(searchStr);
+  } else {
+    searchArray.push(e.key);
+  }
+}
+
+function findMatch(searchStr) {
+  text = text.replaceAll(
+    searchStr,
+    "<span class='highlight'>" + searchStr + "</span>"
```

Comment:
With this method if I search for any letter in the <span class ='highlight'> string and then press shift again, it modifies the text to include span class highlight as a string.